src/App.tsx

```diff
@@ -1,24 +1,197 @@
+      const updatedTodo = await updateTodo(todoId, newTodoData);
+
+      setTodos(prevTodos => prevTodos.map(todo => {
+        if (todo.id !== todoId) {
+          return todo;
+        }
+
+        return updatedTodo;
+      }));
+    } catch {
+      setError('Unable to update a todo');
+    } finally {
+      setLoadingTodos(prevIds => prevIds.filter(id => id !== todoId));
+    }
+  };
+
+  const handleToggleAllButton = () => {
+    todos.forEach(async (todo) => {
+      await updateTodoInfo(
+        todo.id,
+        { completed: !isAllTodosCompleted },
+      );
+    });
+  };
+
+  const deleteTodo = async (todoId: number) => {
+    try {
+      setLoadingTodos(prevIds => [...prevIds, todoId]);
+      await removeTodo(todoId);
+      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setError('Unable to delete a todo');
+    } finally {
+      setLoadingTodos(prevIds => prevIds.filter(id => id !== todoId));
+    }
+  };
+
+  const handleDeleteCompletedButton = async () => {
+    const deletePromises = completedTodos.map(todo => deleteTodo(todo.id));
+
+    try {
+      await Promise.all(deletePromises);
+    } catch {
+      setError('Unable to delete todos');
+    }
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+
+          <button
+            type="button"
+            className={cn('todoapp__toggle-all', {
+              active: isAllTodosCompleted,
+            })}
+            onClick={handleToggleAllButton}
+          />
+          <TodoForm setError={setError} addTodo={addTodo} tempTodo={tempTodo} />
+        </header>
+
+        <TodoList
+          todos={visibleTodos}
+          tempTodo={tempTodo}
+          loadingTodos={loadingTodos}
+          deleteTodo={deleteTodo}
+          updateTodo={updateTodoInfo}
+        />
+
+        {todos.length > 0 && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${activeTodos.length} items left`}
+            </span>
+
+            <Filter filter={filter} setFilter={setFilter} />
+
+            {completedTodos.length > 0 && (
+              <button
+                type="button"
+                className="todoapp__clear-completed"
+                onClick={handleDeleteCompletedButton}
+              >
+                Clear completed
+              </button>
+            )}
```

Comment:
Change logic to hide this button by `visibility` to avoid this jumps

![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/6f8b8be7-5058-4dd1-857d-fdbda47caf11)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/419d3456-d20d-407b-b037-9d945f3e1a23)